File: check_jsx.js
const fs = require('fs');
const parser = require('@babel/parser');
const content = fs.readFileSync('C:/AS/AIStory/frontend/src/pages/editor/components/SubjectLibrary.jsx', 'utf8');

try {
  parser.parse(content, {
    sourceType: 'module',
    plugins: ['jsx']
  });
  console.log('No syntax errors found by babel.');
} catch (e) {
  console.error(e.message);
  console.log('Error at location:', e.loc);
}

File: check_jsx.py

with open(r'c:\AS\AIStory\frontend\src\pages\editor\components\ScriptEditor.jsx', 'r', encoding='utf-8') as f:
    text = f.read()

idx = text.find('return (')
if idx != -1:
    print(text[idx:idx+2500])
else:
    print('Not found')

